sympy/simplify/simplify.py

```diff
@@ -649,6 +652,7 @@ def sum_simplify(s):
     from sympy.concrete.summations import Sum
     from sympy.core.function import expand
 
+    s = extract_constants_from_sum_integral(s)
```

Comment:
OK, now this line goes to the first line of `Sum._eval_simplify` and then the `extract_constants_from_sum_integral` routine could go into `concrete/expr_with_limits.py` and then no changes would be necessary in the simplify.py file.

Reply:
```
In [2]: simplify(Integral(x*y, (x, 1, n))+ Sum(x*y, (x, 1, n)))                 
Out[2]: 
    n               
   ___              
   ╲        n       
    ╲   x   ⌠       
y⋅  ╱     + ⎮ x⋅y dx
   ╱        ⌡       
   ‾‾‾      1       
  x = 1           
```
I think If we have to do this then it will not work as expected.
because of this line     https://github.com/sympy/sympy/blob/b68caad822d3ad0ae1fa09c78c9ed2db927e73cb/sympy/simplify/simplify.py#L593
it will go into https://github.com/sympy/sympy/blob/b68caad822d3ad0ae1fa09c78c9ed2db927e73cb/sympy/simplify/simplify.py#L643

Reply:
Ahh. How about
````python
    if expr.has(Sum, Integral):
        expr = extract_constants_from_sum_integral(expr)
        if expr.has(Sum):
            expr = sum_simplify(expr)
```